src/main/java/edu/harvard/iq/dataverse/api/AbstractApiBean.java

```diff
@@ -322,7 +337,24 @@ protected AuthenticatedUser getRequestAuthenticatedUserOrDie(ContainerRequestCon
         if (requestUser.isAuthenticated()) {
             return (AuthenticatedUser) requestUser;
         } else {
-            throw new WrappedResponse(authenticatedUserRequired());
+            final UserRecordIdentifier userRecordIdentifier = oidcLoginBackingBean.getUserRecordIdentifier();
+            if (userRecordIdentifier == null) {
+                AuthenticationStatus status = securityContext.authenticate(httpRequest, httpResponse, null);
+                if (AuthenticationStatus.SUCCESS.equals(status)) {
+                    try {
+                        return (AuthenticatedUser) httpRequest.getAttribute(ApiConstants.CONTAINER_REQUEST_CONTEXT_USER);
+                    } catch (Exception e) {
+                        throw new WrappedResponse(authenticatedUserRequired());
+                    }
+                }
+                throw new WrappedResponse(authenticatedUserRequired());
+            }
+            final AuthenticatedUser authUser = authSvc.lookupUser(userRecordIdentifier);
+            if (authUser != null) {
+                return authUser;
+            } else {
+                throw new WrappedResponse(authenticatedUserRequired());
+            }
```

Comment:
This is hacky. I commented about my view on this at other places. We should discuss this further.